diff -Naur tcl8.5.8.orig/unix/Makefile.in tcl8.5.8/unix/Makefile.in
--- tcl8.5.8.orig/unix/Makefile.in	2009-11-13 18:38:45.000000000 +0000
+++ tcl8.5.8/unix/Makefile.in	2010-02-16 01:45:31.000000000 +0000
@@ -745,7 +745,7 @@
 	fi
 	@EXTRA_INSTALL_BINARIES@
 
-install-libraries: libraries $(INSTALL_TZDATA) install-msgs
+install-libraries: libraries $(INSTALL_TZDATA)
 	@for i in "$(INCLUDE_INSTALL_DIR)" "$(SCRIPT_INSTALL_DIR)"; \
 	    do \
 	    if [ ! -d "$$i" ] ; then \
