diff -Naur util-linux-2.12a/login-utils/chfn.c util-linux-2.12a-new/login-utils/chfn.c
--- util-linux-2.12a/login-utils/chfn.c	2004-03-04 12:13:37.000000000 -0800
+++ util-linux-2.12a-new/login-utils/chfn.c	2004-06-09 00:21:42.862301088 -0700
@@ -200,7 +200,7 @@
 # else /* USE_PAM */
     /* require password, unless root */
     if(uid != 0 && oldf.pw->pw_passwd && oldf.pw->pw_passwd[0]) {
-	char *pwdstr = getpass(_("Password: "));
+	char *pwdstr = getpass(_(""));
 	if(strncmp(oldf.pw->pw_passwd,
 		   crypt(pwdstr, oldf.pw->pw_passwd), 13)) {
 	    puts(_("Incorrect password."));
diff -Naur util-linux-2.12a/login-utils/chsh.c util-linux-2.12a-new/login-utils/chsh.c
--- util-linux-2.12a/login-utils/chsh.c	2004-03-04 12:13:37.000000000 -0800
+++ util-linux-2.12a-new/login-utils/chsh.c	2004-06-09 00:21:47.177645056 -0700
@@ -191,7 +191,7 @@
 # else /* USE_PAM */
     /* require password, unless root */
     if(uid != 0 && pw->pw_passwd && pw->pw_passwd[0]) {
-	char *pwdstr = getpass(_("Password: "));
+	char *pwdstr = getpass(_(""));
 	if(strncmp(pw->pw_passwd,
 		   crypt(pwdstr, pw->pw_passwd), 13)) {
 	    puts(_("Incorrect password."));
diff -Naur util-linux-2.12a/login-utils/cryptocard.c util-linux-2.12a-new/login-utils/cryptocard.c
--- util-linux-2.12a/login-utils/cryptocard.c	2001-03-15 02:09:58.000000000 -0800
+++ util-linux-2.12a-new/login-utils/cryptocard.c	2004-06-09 00:21:52.433845992 -0700
@@ -188,7 +188,7 @@
     if (challenge == NULL) return 0;
 
     if (strlen(challenge) + 13 > sizeof(prompt)) return 0;
-    sprintf(prompt, "%s Password: ", challenge);
+    sprintf(prompt, "%s", challenge);
 
     alarm((unsigned int)timeout);  /* give user time to fiddle with card */
     response = getpass(prompt);  /* presents challenge and gets response */
diff -Naur util-linux-2.12a/login-utils/login.c util-linux-2.12a-new/login-utils/login.c
--- util-linux-2.12a/login-utils/login.c	2004-03-04 11:35:34.000000000 -0800
+++ util-linux-2.12a-new/login-utils/login.c	2004-06-09 00:22:01.471472064 -0700
@@ -771,7 +771,7 @@
 	  break;
 	
 	setpriority(PRIO_PROCESS, 0, -4);
-	pp = getpass(_("Password: "));
+	pp = getpass(_(""));
 	
 #  ifdef CRYPTOCARD
 	if (strncmp(pp, "CRYPTO", 6) == 0) {
diff -Naur util-linux-2.12a/login-utils/newgrp.c util-linux-2.12a-new/login-utils/newgrp.c
--- util-linux-2.12a/login-utils/newgrp.c	2001-03-15 02:09:58.000000000 -0800
+++ util-linux-2.12a-new/login-utils/newgrp.c	2004-06-09 00:22:13.021716160 -0700
@@ -45,7 +45,7 @@
 
     if(ge->gr_passwd && ge->gr_passwd[0] != 0) {
 	if(strcmp(ge->gr_passwd, 
-		  crypt(getpass(_("Password: ")), ge->gr_passwd)) == 0) {
+		  crypt(getpass(_("")), ge->gr_passwd)) == 0) {
 	    return TRUE;		/* password accepted */
 	}
     }
diff -Naur util-linux-2.12a/login-utils/simpleinit.c util-linux-2.12a-new/login-utils/simpleinit.c
--- util-linux-2.12a/login-utils/simpleinit.c	2002-11-24 14:31:53.000000000 -0800
+++ util-linux-2.12a-new/login-utils/simpleinit.c	2004-06-09 00:21:56.265263528 -0700
@@ -335,7 +335,7 @@
 
     for (i = 0; i < MAXTRIES; i++)
     {
-	pass = getpass (_ ("Password: ") );
+	pass = getpass (_ ("") );
 	if (pass == NULL) continue;
 		
 	if ( !strcmp (crypt (pass, rootpass), rootpass) ) return 1;
diff -Naur util-linux-2.12r-orig/mount/lomount.c util-linux-2.12r/mount/lomount.c
--- util-linux-2.12r-orig/mount/lomount.c	2006-07-30 13:14:35.000000000 -0700
+++ util-linux-2.12r/mount/lomount.c	2006-07-30 13:15:44.822925750 -0700
@@ -472,7 +472,7 @@
 	int i, ask2;
 
 	if(!passFDnumber) {
-		p = getpass(_("Password: "));
+		p = getpass(_(""));
 		ask2 = passAskTwice ? 1 : 0;
 	} else {
 		i = atoi(passFDnumber);
