diff -Naur util-linux-2.12r.orig/disk-utils/fsck.cramfs.c util-linux-2.12r/disk-utils/fsck.cramfs.c
--- util-linux-2.12r.orig/disk-utils/fsck.cramfs.c	2004-12-11 08:53:16.000000000 -0600
+++ util-linux-2.12r/disk-utils/fsck.cramfs.c	2008-05-27 00:42:01.000000000 -0500
@@ -76,7 +76,6 @@
 
 #define PAD_SIZE 512
 
-#include <asm/page.h>
 #ifdef PAGE_SIZE
 #define PAGE_CACHE_SIZE ((int) PAGE_SIZE)
 #elif defined __ia64__
