Index: avidemux-2.5.5/plugins/ADM_videoEncoder/ADM_vidEnc_x264/encoder.cpp
===================================================================
--- avidemux-2.5.5.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/encoder.cpp	2011-06-07 08:37:03.127997243 +0200
+++ avidemux-2.5.5/plugins/ADM_videoEncoder/ADM_vidEnc_x264/encoder.cpp	2011-06-07 08:38:06.281464832 +0200
@@ -488,8 +488,10 @@
 	printf("[x264] analyse.b_transform_8x8 = %d\n", x264Param->analyse.b_transform_8x8);
 	printf("[x264] analyse.inter = %d\n", x264Param->analyse.inter);
 	printf("[x264] b_cabac = %d\n", x264Param->b_cabac);
-#if X264_BUILD > 101
+#if X264_BUILD > 101 && X264_BUILD < 115
 	printf("[x264] i_open_gop = %d\n", x264Param->i_open_gop);
+#elif X264_BUILD > 114
+	printf("[x264] b_open_gop = %d\n", x264Param->b_open_gop);
 #endif
 	printf("[x264] b_interlaced = %d\n", x264Param->b_interlaced);
 #if X264_BUILD > 88
Index: avidemux-2.5.5/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp
===================================================================
--- avidemux-2.5.5.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp	2011-06-07 08:38:15.341100944 +0200
+++ avidemux-2.5.5/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp	2011-06-07 08:41:08.114138787 +0200
@@ -417,7 +417,7 @@
 	_param.b_cabac = cabac;
 }
 
-#if X264_BUILD > 101
+#if X264_BUILD > 101 && X264_BUILD < 115
 unsigned int x264Options::getOpenGopMode(void)
 {
 	return _param.i_open_gop;
@@ -428,6 +428,16 @@
 	if (openGopMode < 3)
 		_param.i_open_gop = openGopMode;
 }
+#elif X264_BUILD > 114
+unsigned int x264Options::getOpenGopMode(void)
+{
+	return _param.b_open_gop;
+}
+void x264Options::setOpenGopMode(unsigned int openGopMode)
+{
+	if (openGopMode < 3)
+		_param.b_open_gop = openGopMode;
+}
 #endif
 
 unsigned int x264Options::getInterlaced(void)
