diff -Naur gnome-utils-3.2.1.orig/Makefile.am gnome-utils-3.2.1/Makefile.am
--- gnome-utils-3.2.1.orig/Makefile.am	2011-09-27 14:27:36.000000000 -0700
+++ gnome-utils-3.2.1/Makefile.am	2011-12-02 16:26:08.115467307 -0800
@@ -21,13 +21,8 @@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
 SUBDIRS = \
-	m4			\
-	po			\
 	libeggsmclient		\
 	baobab			\
-	font-viewer		\
-	gnome-dictionary	\
-	gnome-screenshot	\
 	gsearchtool		\
 	logview			\
 	$(NULL)
diff -Naur gnome-utils-3.2.1.orig/Makefile.in gnome-utils-3.2.1/Makefile.in
--- gnome-utils-3.2.1.orig/Makefile.in	2011-10-17 07:29:45.000000000 -0700
+++ gnome-utils-3.2.1/Makefile.in	2011-12-02 16:26:17.754638002 -0800
@@ -320,13 +320,8 @@
 distcleancheck_listfiles = find . -type f -print | grep -v 'omf\.out' | grep -v 'legal.xml'
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 SUBDIRS = \
-	m4			\
-	po			\
 	libeggsmclient		\
 	baobab			\
-	font-viewer		\
-	gnome-dictionary	\
-	gnome-screenshot	\
 	gsearchtool		\
 	logview			\
 	$(NULL)
diff -Naur gnome-utils-3.2.1.orig/configure gnome-utils-3.2.1/configure
--- gnome-utils-3.2.1.orig/configure	2011-10-17 07:29:42.000000000 -0700
+++ gnome-utils-3.2.1/configure	2011-12-02 16:08:44.255564285 -0800
@@ -19415,12 +19415,12 @@
 fi
 
 
-if test x$SCROLLKEEPER_CONFIG = xno; then :
-
-        as_fn_error $? "Couldn't find scrollkeeper-config. Please install the scrollkeeper package" "$LINENO" 5
-
-
-fi
+#if test x$SCROLLKEEPER_CONFIG = xno; then :
+#
+#        as_fn_error $? "Couldn't find scrollkeeper-config. Please install the scrollkeeper package" "$LINENO" 5
+#
+#
+#fi
 
 
 # use strict compiler flags only on development releases
diff -Naur gnome-utils-3.2.1.orig/configure.ac gnome-utils-3.2.1/configure.ac
--- gnome-utils-3.2.1.orig/configure.ac	2011-10-17 07:14:53.000000000 -0700
+++ gnome-utils-3.2.1/configure.ac	2011-12-02 16:17:47.548866221 -0800
@@ -31,8 +31,6 @@
 AC_PATH_PROG(GLIB_GENMARSHAL, [glib-genmarshal])
 AC_PATH_PROG(GLIB_MKENUMS, [glib-mkenums])
 
-GNOME_COMPILE_WARNINGS([maximum])
-
 AC_PATH_XTRA
 
 AC_TYPE_OFF_T
@@ -290,12 +288,12 @@
 AC_SUBST(Z_LIBS)
 
 dnl scrollkeeper checks 
-AC_PATH_PROG(SCROLLKEEPER_CONFIG, [scrollkeeper-config], [no])
-AS_IF([test x$SCROLLKEEPER_CONFIG = xno],
-      [
-        AC_MSG_ERROR([Couldn't find scrollkeeper-config. Please install the scrollkeeper package])
-      ]
-)
+#AC_PATH_PROG(SCROLLKEEPER_CONFIG, [scrollkeeper-config], [no])
+#AS_IF([test x$SCROLLKEEPER_CONFIG = xno],
+#      [
+#        AC_MSG_ERROR([Couldn't find scrollkeeper-config. Please install the scrollkeeper package])
+#      ]
+#)
 
 dnl = Enable strict compiler flags =========================================
 
