diff -Naur linux-2.6.34.orig/arch/x86/include/asm/percpu.h linux-2.6.34/arch/x86/include/asm/percpu.h
--- linux-2.6.34.orig/arch/x86/include/asm/percpu.h	2010-05-16 14:17:36.000000000 -0700
+++ linux-2.6.34/arch/x86/include/asm/percpu.h	2012-03-08 16:44:43.354589685 -0800
@@ -77,6 +77,7 @@
 	if (0) {					\
 		pto_T__ pto_tmp__;			\
 		pto_tmp__ = (val);			\
+		(void)pto_tmp__;			\
 	}						\
 	switch (sizeof(var)) {				\
 	case 1:						\
@@ -115,6 +116,7 @@
 	if (0) {							\
 		pao_T__ pao_tmp__;					\
 		pao_tmp__ = (val);					\
+		(void)pao_tmp__;					\
 	}								\
 	switch (sizeof(var)) {						\
 	case 1:								\
