--- mountfs.orig	2007-12-19 23:53:23.000000000 +0000
+++ mountfs	2008-01-20 16:08:22.000000000 +0000
@@ -21,6 +21,36 @@
 		mount -n -o remount,rw / >/dev/null
 		evaluate_retval
 
+#
+# Begin uproot code
+#
+
+UPBALL=/mnt/boot/root-deltas.tbz 
+
+        for i in `cat /proc/cmdline`; do
+            case $i in
+                uproot=*)
+                    UPDEV=`echo $i| sed 's/^.*=//'` 
+                    mount -t vfat $UPDEV /mnt
+
+                    if [ $? -ne 0 ]; then
+                        mount -t auto $UPDEV /mnt
+                    fi
+
+                    if [ -f $UPDDEV ]; then
+                        # prevent prehistoric timestamps
+                        /etc/rc.d/init.d/setclock start
+                        tar jxvmf $UPBALL -C /
+                    fi
+
+                    umount /mnt
+                    ;;
+            esac
+        done
+#
+# End uproot code
+#
+
 		# Remove fsck-related file system watermarks.
 		rm -f /fastboot /forcefsck
 
