diff -Naur tunctl-1.5.orig/Makefile tunctl-1.5/Makefile
--- tunctl-1.5.orig/Makefile	2008-07-09 15:31:52.000000000 -0700
+++ tunctl-1.5/Makefile	2012-01-26 17:46:27.000000000 -0800
@@ -11,7 +11,7 @@
 BIN_DIR ?= /usr/sbin
 MAN_DIR ?= /usr/share/man/man$(MANS)
 
-all : $(BIN) $(MAN)
+all : $(BIN)
 
 $(BIN) : $(BIN).c
 	$(CC) $(CFLAGS) -o $(BIN) $(BIN).c
@@ -22,11 +22,11 @@
 clean : 
 	rm -f $(BIN) $(OBJS) $(MAN) *~ manpage.*
 
-install : $(BIN) $(MAN)
+install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
 	install $(BIN) $(DESTDIR)$(BIN_DIR)
-	install -d $(DESTDIR)$(MAN_DIR)
-	install $(MAN) $(DESTDIR)$(MAN_DIR)
+	echo install -d $(DESTDIR)$(MAN_DIR)
+	echo install $(MAN) $(DESTDIR)$(MAN_DIR)
 
 .PHONY: dist
 dist: distcheck
